docs/js-guide/dependencies.rst

```diff
@@ -1,72 +1,145 @@
+    import "commcarehq";  // REQUIRED at the top of every "entry point"
+                          // This loads site-wide dependencies needed to run global navigation, modals, notifications, etc.
 
-You are likely working with RequireJS, as most of HQ has been migrated.
-However, several major areas have **not** been migrated: app manager,
-reports, and web apps. Test code also does not currently use RequireJS;
-see
-`Testing <https://github.com/dimagi/commcare-hq/blob/master/docs/js-guide/testing.rst>`__
-for working with tests.
+    // Common third-party dependencies
+    import $ from "jquery";                     // Ideally, new pages should move away from jQuery and use native
+                                                // javascript. But this is here as an example.
+    import ko from "knockout";
+    import "hqwebapp/js/knockout_bindings.ko";  // This one doesn't need a named parameter because it only adds
+                                                // knockout bindings and is not referenced in this file
+    import _ from "underscore";
+
+    // A commonly used internal module for passing server-side data to the front end
+    import initialPageData from "hqwebapp/js/initial_page_data";
+
+    /* page-level entry point logic begins here */
+
+
+
+To register your module as a Webpack entry point, add the ``webpack_main`` template tag to your HTML template,
+near the top and outside of any other block:
 
-To tell for sure, look at your module’s ``hqDefine`` call, at the top of
-the file.
+::
+
+   {% webpack_main 'prototype/js/example' %}
+
+Some pages don't have any unique logic but do rely on other modules.
+These are usually pages that use some common widgets but don't have custom UI interactions.
+
+If your page only relies on a single JavaScript module, you can use that as that
+page's entry point:
+
+::
+
+   {% webpack_main 'locations/js/widgets' %}
+
+If your page relies on multiple modules, it still needs one entry point.
+You can handle this by making a module that only imports other modules.
+For instance an entry point located at ``prototype/js/combined_example.js``
+might look like:
+
+::
+
+    import "commcarehq";  // always at the top
+
+    import "hqwebapp/js/crud_paginated_list_init";
+    import "hqwebapp/js/bootstrap5/widgets";
+
+    // No page-specific logic, just need to collect the dependencies above
+
+Then in your HTML page:
+
+::
 
-RequireJS modules look like this, with all dependencies loaded as part
-of ``hqDefine``:
+   {% webpack_main 'prototype/js/combined_example' %}
+
+The exception to the above is if your page inherits from a legacy page that
+doesn't use a JavaScript bundler. This is rare, but one example would be adding a
+new page to app manager that inherits from ``managed_app.html``.
+
+
+Why is old code formatted differently?
+--------------------------------------
+
+You may notice that older JavaScript code on HQ is written in a modified AMD
```

Comment:
i like always stating that it is a modification of a global standard rather than something we made up
happy to also say "modified AMD (``hqDefine``)"

Reply:
Using both sounds good, I agree there's value to explaining which parts are Dimagi-specific and which aren't.

Reply:
also, there is a difference between modified and transitional, because the transitional models do not have the list of dependencies and allow `hqImport` in the body...that's the distinction I was trying to make with this differentiation

Reply:
Hmm, I didn't get that distinction while reading this. What if instead of modified the docs just said "AMD", or "wrapped AMD", since for these `hqDefine` is a thin wrapper around `define`? And for transitional, maybe something like "non-bundled"?

I think in my head, both types of modules are intermediate states, so "transitional" didn't click for me.